tls/s2n_server_hello.c

```diff
@@ -273,6 +273,9 @@ int s2n_server_hello_recv(struct s2n_connection *conn)
 
     conn->actual_protocol_version_established = 1;
 
+    /* At this point, we have selected a state machine for this handshake */
+    conn->is_tls13_state_machine = (conn->actual_protocol_version == S2N_TLS13);
```

Comment:
You mentioned in https://github.com/aws/s2n-tls/pull/3689/files#r1041353851 that this could be set in `s2n_conn_set_tls13_handshake_type`. I wonder if this would be worth doing, since this is called when receiving both the server hello and the client hello, so this value would only have to be set in one place?

Reply:
I'm not sure if this saves us much, although it would be a little cleaner.